data/data/azure/bootstrap/main.tf

```diff
@@ -146,7 +152,8 @@ resource "azurerm_virtual_machine" "bootstrap" {
   }
 
   depends_on = [
-    azurerm_network_interface_backend_address_pool_association.public_lb_bootstrap,
+    # XXX: need to figure out how to do this in terraform
+    #gazurerm_network_interface_backend_address_pool_association.public_lb_bootstrap,
```

Comment:
i think you can keep depend on even if the count is 0 for that resource, are you seeing any error with that?

Reply:
I've removed the comment, but I believe it errors out. I'm running again soon to confirm.

Reply:
Confirmed that it fails.

Reply:
can you attach the exact error to the commit message??

there seems to be some information here:
https://github.com/hashicorp/terraform/issues/15285#issuecomment-447971852

Reply:
ERROR                                              
ERROR Error: Reference to undeclared resource      
ERROR                                              
ERROR   on ../../../tmp/openshift-install-412731445/bootstrap/main.tf line 155, in resource "azurerm_virtual_machine" "bootstrap": 
ERROR  155:     gazurerm_network_interface_backend_address_pool_association.public_lb_bootstrap, 
ERROR                                              
ERROR A managed resource                           
ERROR "gazurerm_network_interface_backend_address_pool_association" 
ERROR "public_lb_bootstrap" has not been declared in bootstrap. 
ERROR                                              
ERROR Failed to read tfstate: open /tmp/openshift-install-412731445/terraform.tfstate: no such file or directory 
FATAL failed to fetch Cluster: failed to generate asset "Cluster": failed to create cluster: failed to apply using Terraform

Reply:
<pre>
ERROR                                              
ERROR Error: Reference to undeclared resource      
ERROR                                              
ERROR   on ../../../tmp/openshift-install-412731445/bootstrap/main.tf line 155, in resource "azurerm_virtual_machine" "bootstrap": 
ERROR  155:     gazurerm_network_interface_backend_address_pool_association.public_lb_bootstrap, 
ERROR                                              
ERROR A managed resource                           
ERROR "gazurerm_network_interface_backend_address_pool_association" 
ERROR "public_lb_bootstrap" has not been declared in bootstrap. 
ERROR                                              
ERROR Failed to read tfstate: open /tmp/openshift-install-412731445/terraform.tfstate: no such file or directory 
FATAL failed to fetch Cluster: failed to generate asset "Cluster": failed to create cluster: failed to apply using Terraform 
</pre>

Reply:
@jhixson74 would that might be because `gazurerm_network_interface_backend_address_pool_association` has an extra g in the front of the resource name.